app/Filament/Resources/UserResource/Pages/Roles.php

```diff
@@ -53,6 +53,15 @@ public function table(Table $table): Table
 
                         $attachedRole = Role::findById((int) $data['recordId']);
 
+                        if (!in_array($attachedRole->name, [
+                            Role::DEVELOPER_JUNIOR,
+                            Role::DEVELOPER,
+                            Role::DEVELOPER_STAFF,
+                            Role::DEVELOPER_RETIRED,
+                        ])) {
+                            return;
+                        }
```

Comment:
Having this as a secondary gate duplicates the lower logic and creates an opportunity for them to become desync'd. Suggest adding an `else return` to the logic below:
```
else
{
    // nothing to synchronize for non-developer roles. we're done
    return;
}
```

Reply:
Addressed in latest.

Reply:
I think you missed my point. You're still checking the conditions twice.

Instead of:
```
if (in_array(A, B)) {
    if (A) {
    } elseif (B) {
    }
    do_stuff();
}
```
just do:
```
if (A) {
} elseif (B) {
} else {
    return;
}
do_stuff();
```

Reply:
Updated in latest.